src/handler/sign_tx.c

```diff
@@ -197,8 +227,27 @@ void read_tx_data(buffer_t *cdata) {
           buffer_read_u8(cdata, &G_context.tx_info.remaining_inputs) &&
           buffer_read_u8(cdata, &G_context.tx_info.outputs_len))) {
         // if an error occurs reading
-        THROW(SW_WRONG_DATA_LENGTH);
+        return SW_WRONG_DATA_LENGTH;
+    }
+
+    for (uint8_t i = 0; i < G_context.tx_info.change_len; i++) {
+        // check that all change indexes are inside the outputs array
+        if (G_context.tx_info.change_info[i].index >= G_context.tx_info.outputs_len) {
```

Comment:
I'd like to confirm that we confirm that all change outputs have addresses that belong to this wallet.

Reply:
We have a method `verify_change` (under `src/handler/sign_tx.c)` that checks this and runs for all change outputs on the transaction.

Reply:
Thanks!